examples/url_scan_report.py

```diff
@@ -0,0 +1,104 @@
+"""
+This example program allows a user to pass a URL as a command line argument during script execution.
+
+EXAMPLE: python3 url_scan_report.py testphp.vulnweb.com
+
+This will return the list of security vendors who have marked the URL as clean or malicious, 
+and can be used to score the trust worthiness of a URL.
+
+NOTICE: In order to use this program you will need an API key that has
+privileges for using the VirusTotal Feed API.
+
+Developed by @David-M-Berry
+"""
+
+import argparse
+import os
+import time
+import vt
+
+def scan_url(client, url):
+    """Initiates a scan for the given URL and returns the scan ID."""
+    print(f"Scanning URL: {url}")
+    try:
+        analysis = client.scan_url(url)
+        print(f"URL scan started: {analysis.id}")
+        return analysis.id
+    except vt.error.APIError as e:
+        print(f"APIError: {e}")
+    except Exception as e:
+        print(f"An error occurred during URL scan: {e}")
+    return None
+
+def get_url_report(client, scan_id):
+    """Fetches the report for the given scan ID."""
+    print(f"Fetching report for scan ID: {scan_id}")
+    try:
+        # Fetch the analysis using the scan ID
+        analysis = client.get_object(f"/analyses/{scan_id}")
+        return analysis
+    except vt.error.APIError as e:
+        print(f"APIError: {e}")
+    except Exception as e:
+        print(f"An error occurred while fetching the report: {e}")
+    return None
+
+def wait_for_scan_completion(client, scan_id, timeout=300):
+    """Waits for the scan to complete by checking the status periodically."""
+    print("Waiting for the scan to complete...")
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        report = get_url_report(client, scan_id)
```

Comment:
The Client object has a timeout option that can be used instead of doing this:
https://github.com/VirusTotal/vt-py/blob/03fd2f1a4ee36debdb717a7c0d9329de081d0b0c/vt/client.py#L234